tools/cloud-build/daily-tests/ansible_playbooks/slurm-integration-test.yml

```diff
@@ -184,43 +203,59 @@
 
     ## Always cleanup, even on failure
     always:
+    - name: Ensure all nodes are powered down
+      ansible.builtin.command: sinfo -t 'POWERED_DOWN' --noheader --format "%n"
+      register: final_node_count
+      changed_when: False
+      until: final_node_count.stdout_lines | length == initial_node_count.stdout_lines | length
```

Comment:
This will break unless initial_node_count is acquired somehow before, which only happens with test partitions. If you want to move this into the slurm-integration-test, then you have to move the following *before* running the integration tests: 
```
- name: Count Slurm nodes
  ansible.builtin.shell:
    sinfo -t 'IDLE&POWERED_DOWN' --noheader --format "%n"
  args:
    executable: /bin/bash
  changed_when: False
  register: initial_node_count
  
  ```

Reply:
Updated.